### PR TITLE
cifsd: fix side effects (smb2.create)

### PR DESCRIPTION
**Pull request by tymok**
fix side effects (smb2.create.mkdir-dup, multi)

to avoid error return in smb2.create.mkdir-dup,
check NT_STATUS_NOT_A_DIRECTORY only if named stream exists.
smb2.create.mkdir-dup gives requests whose FileAttributes is 0x80
and CreateOptions is FILE_DIRECTORY_FILE.

modify NT_STATUS_OBJECT_NAME_COLLISION condition in order to pass
smb2.create.multi that sends multiple create requests
with same names and CreateOptions=0.

Signed-off-by: Taeyang Mok <t.mok@samsung.com>

### CODE REVIEW

File: smb2pdu.c [hunk 1]
```diff
@@ -2155,6 +2148,13 @@ int smb2_open(struct smb_work *smb_work)
 				goto err_out;
 			}
 		}
+
+		if (req->CreateOptions & FILE_DIRECTORY_FILE_LE &&
```
Comment (namjaejeon) [inline smb2pdu.c]: Is there any reason why this check move to here ?

File: smb2pdu.c [hunk 2]
```diff
@@ -2155,6 +2148,13 @@ int smb2_open(struct smb_work *smb_work)
...
+			req->FileAttributes & FILE_ATTRIBUTE_NORMAL_LE) {
+			rsp->hdr.Status = NT_STATUS_NOT_A_DIRECTORY;
+			rc = -EIO;
+			goto err_out1;
```
Comment (namjaejeon) [inline smb2pdu.c]: looks incorrect goto, need to change err_out1 => err_out

File: smb2pdu.c (hunk 1 above)
Comment (tymok) [reply smb2pdu.c]: As we discussed, this statement is for smb2.streams TC.
By the way, smb2.create.mkdir-dup gives requests whose FileAttributes is FILE_ATTRIBUTE_NORMAL and CreateOptions is FILE_DIRECTORY_FILE.
So I think It's better to make this statement runs only on the streams cases.

File: smb2pdu.c (hunk 2 above)
Comment (tymok) [reply smb2pdu.c]: Thank you for correcting the error.
I have corrected the goto statement as you pointed out. please check the new patch.

**Pull request merged**